release.nix

```diff
@@ -1,22 +1,38 @@
 {
   niv ? import nix/sources.nix,
-  pkgs ? import niv.nixpkgs { config = { allowUnfree = true; }; }
+  nixpkgs ? niv.nixpkgs,
+  pkgs ? import nixpkgs { overlays = [ (import ./overlay.nix) ]; }
 }:
 let
-  cbsLib = import ./lib { inherit pkgs; };
-  cbspkgs = import ./pkgs { inherit niv pkgs cbsLib; };
+  cbsLib = (import ./default.nix { inherit nixpkgs; }).lib;
+  tests = import ./tests { inherit nixpkgs; };
+  libtests = import ./lib/tests { inherit cbsLib; };
+
   flatten = with pkgs.lib; collect isDerivation;
-  tests = import ./tests { inherit (niv) nixpkgs; };
+  attrFilter = n: _: n != "lib" && n != "modules" && n != "nixpkgs";
+
+  # somehow hydra does not like the `recurseForDerivations = true;` attributes.
+  # it says "in job ‘randomAttr.recurseForDerivations’: unsupported value: true"
+  # if these are not filtered out. It remains to be researched why this does not
+  # seem to be necessary in nixpkgs where such attributes are used, too.
+  removeRecurseLabels = x: with builtins; let
+    filteredTree = listToAttrs (
+      map
+        (n: pkgs.lib.nameValuePair n (removeRecurseLabels x.${n}))
+        (filter (n: n != "recurseForDerivations") (attrNames x))
+    );
```

Comment:
Is there any difference to `filterAttrsRecursive (n: _: n != "recurseForDerivations") x`?

Reply:
Would  `dontRecureseForAttrs` suitable too ?  
If i understand it correctly it sets `recurseForDerivations` to `false`

Reply:
unfortunately, no. the problem is that hydra itself does not like it if you present it a tree of attribute sets where some values are not derviations but `bool` or something. So they have to be removed completely instead of being set to `false`.